src/java.base/share/classes/jdk/internal/org/objectweb/asm/Type.java

```diff
@@ -668,11 +668,10 @@ private static void appendDescriptor(final Class<?> clazz, final StringBuilder s
             stringBuilder.append(descriptor);
         } else {
             String name = currentClass.getName();
-            if (Helper.isIndirectType(currentClass)) {
-                stringBuilder.append('L');
-            } else {
+            if (isInlineClass(currentClass)) {
                 stringBuilder.append('Q');
-
+            } else {
+                stringBuilder.append('L');
```

Comment:
This could be:
`stringBuilder.append(IsInlineClass(currentClass) ? 'Q' : 'L');`

Reply:
`Class::descriptorString` should be used for the cases when it has a `Class` object.

`ProxyGenerator` and `AccessorGenerator` should be updated to use `Class::descriptorString` and `TypeDescriptor` API where it's appropriate.  I suggest to update these classes in jdk/jdk.  I will file a JBS issue to track this.

Reply:
```
diff --git a/src/java.base/share/classes/jdk/internal/org/objectweb/asm/Type.java b/src/java.base/share/classes/jdk/internal/org/objectweb/asm/Type.java
index 680e764fb4c..b03c201e23e 100644
--- a/src/java.base/share/classes/jdk/internal/org/objectweb/asm/Type.java
+++ b/src/java.base/share/classes/jdk/internal/org/objectweb/asm/Type.java
@@ -667,12 +667,8 @@ public final class Type {
             }
             stringBuilder.append(descriptor);
         } else {
+            stringBuilder.append(isInlineClass(currentClass) ? 'Q' : 'L');
             String name = currentClass.getName();
-            if (isInlineClass(currentClass)) {
-                stringBuilder.append('Q');
-            } else {
-                stringBuilder.append('L');
-            }
             int nameLength = name.length();
             for (int i = 0; i < nameLength; ++i) {
                 char car = name.charAt(i);
```